code/datums/components/sticky.dm

```diff
@@ -69,6 +71,26 @@
 		START_PROCESSING(SSobj, src)
 	return COMPONENT_CANCEL_ATTACK_CHAIN
 
+/datum/component/sticky/proc/stick_to_it_throwing(obj/item/I, atom/hit_target, throwingdatum, params)
+	SIGNAL_HANDLER
+	if(hit_target.GetComponent(/datum/component/sticky))
+		return
+
+	attached_to = hit_target
+	move_to_the_thing(parent)
+
+	overlay = icon(I.icon, I.icon_state)
+
+	attached_to.add_overlay(overlay, priority = TRUE)
+	I.invisibility = INVISIBILITY_ABSTRACT
+
+	RegisterSignal(attached_to, COMSIG_HUMAN_MELEE_UNARMED_ATTACKBY, PROC_REF(pick_up))
+	RegisterSignal(attached_to, COMSIG_PARENT_EXAMINE, PROC_REF(add_sticky_text))
+	RegisterSignal(attached_to, COMSIG_PARENT_QDELETING, PROC_REF(on_attached_destroy))
+	if(ismovable(attached_to))
+		RegisterSignal(attached_to, COMSIG_MOVABLE_MOVED, PROC_REF(on_move))
+		START_PROCESSING(SSobj, src)
```

Comment:
Code duplication. Should extract common code to proc

Reply:
Reviews applied